### PR TITLE
WIP fix(types): transform method only accepts current event type

### PR DESCRIPTION
**Pull request by oscard0m**

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -21,7 +21,10 @@ interface BaseWebhookEvent<TName extends WebhookEventName> {
   payload: WebhookEventMap[TName];
 }
 
-export interface Options<TTransformed = unknown> {
+export interface Options<
+  E extends EmitterWebhookEventName = any,
+  TTransformed extends EmitterWebhookEvent<E> = any
```
Comment (oscard0m) [inline src/types.ts:26]: I would like to improve these `any` default values here. Is it possible?

Comment (wolfy1339) [reply src/types.ts:26]: You can default to `EmitterWebhookEventName`, it's done elsewhere in the code

**Comment by wolfy1339** [comment]
I tested your patch and it doesn't seem to fix the issue at hand

**Comment by wolfy1339** [comment]
I think some next steps would be to check if reverting these PRs fixes the problem
https://github.com/octokit/webhooks.js/pull/459 https://github.com/octokit/webhooks.js/pull/464

**Comment by oscard0m** [comment]
> I think some next steps would be to check if reverting these PRs fixes the problem
> #459 #464

Opened a new PR reverting these 2 PR commits: https://github.com/octokit/webhooks.js/pull/494

Closing this one since it's not fixing the issue, as @wolfy1339 pointed

**Pull request closed without merging**